bin/cylc-subscribe

```diff
@@ -0,0 +1,141 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc subscribe [OPTIONS] ARGS
+
+(This command is for internal use.)
+Invoke suite subscriber to receive published workflow output.
+"""
+
+import asyncio
+import json
+import sys
+import time
+
+from google.protobuf.json_format import MessageToDict
+
+from cylc.flow.option_parsers import CylcOptionParser as COP
+from cylc.flow.network.scan import get_scan_items_from_fs, re_compile_filters
+from cylc.flow.network.subscriber import WorkflowSubscriber, process_delta_msg
+from cylc.flow.terminal import cli_function
+from cylc.flow.ws_data_mgr import DELTAS_MAP
+
+if '--use-ssh' in sys.argv[1:]:
+    sys.argv.remove('--use-ssh')
+    from cylc.flow.remote import remrun
+    if remrun():
+        sys.exit(0)
+
+
+def print_message(topic, data, subscriber=None, once=False):
+    """Print protobuf message."""
+    print(f'Received: {topic}')
+    sys.stdout.write(
+        json.dumps(MessageToDict(data), indent=4) + '\n')
+    if once and subscriber is not None:
+        subscriber.stop()
+
+
+def get_option_parser():
+    """Augment options parser to current context."""
+    parser = COP(
+        __doc__,
+        argdoc=[
+            ('REG', 'Suite name'),
+            ('[USER_AT_HOST]', 'user@host:port, shorthand for --user, '
+             '--host & --port.')],
+        comms=True,
+        noforce=True
+    )
+
+    delta_keys = list(DELTAS_MAP)
+    pb_topics = ("Directly published data-store topics include: '" +
+              ("', '").join(delta_keys[:-1]) +
+              "' and '" + delta_keys[-1] + "'.")
+    parser.add_option(
+        "-T", "--topics",
+        help="Specify a comma delimited list of subscription topics. "
+             + pb_topics,
+        action="store", dest="topics", default='workflow')
+
+    parser.add_option(
+        "-o", "--once",
+        help="Show a single publish then exit.",
+        action="store_true", default=False, dest="once")
+
+
+    return parser
+
+
+@cli_function(get_option_parser)
+def main(_, options, *args):
+    suite = args[0]
+
+    if len(args) > 1:
+        try:
+            user_at_host, options.port = args[1].split(':')
+            options.owner, options.host = user_at_host.split('@')
+        except ValueError:
+            print(('USER_AT_HOST must take the form '
+                   '"user@host:port"'), file=sys.stderr)
+            sys.exit(1)
+    elif options.host is None or options.port is None:
+        cre_owner, cre_name = re_compile_filters(None, ['.*'])
+        while True:
+            for reg, host, _, pub_port in get_scan_items_from_fs(
```

Comment:
Can get contact info for a suite directly (via `cylc.flow.suite_files`) rather than iterating through `cylc scan` results.

https://github.com/cylc/cylc-flow/blob/master/cylc/flow/suite_files.py#L263

Reply:
Right'O .. I'll change to use that or `load_contact_file`

Reply:
Fixed - Still using `get_scan_items_from_fs` (as it returns exactly what I need), but using `suite` as an argument now:
```
cre_owner, cre_name = re_compile_filters(options.owner, [suite])
```

Reply:
The reason I'm worried about this is that `get_scan_items_from_fs` can be quite slow. It's IO bound and some users have *lots* of suites in their `cylc-run` directories. These suites can be nested so Cylc has to do a lot of work walking the FS tree to find items, even if you specify the owner and host it can still be slow.

Reply:
Ok, I see.. `cylc-subscribe` isn't used in the sync BTW, I just used it diagnostically, but it has potential in CLI tools.

Reply:
@oliver-sanders fixed - now uses `load_contact_file` directly.